src/test/java/org/jenkinsci/plugins/build_token_root/BuildRootActionTest.java

```diff
@@ -64,38 +71,40 @@
         testBuild(j.createFreeStyleProject("p"));
     }
 
-    @SuppressWarnings("unchecked") // AbstractItem.getParent snafu
     private <JobT extends Job<JobT, RunT> & ParameterizedJobMixIn.ParameterizedJob, RunT extends Run<JobT, RunT>> void testBuild(JobT p) throws Exception {
+        setAuthToken(p);
         JenkinsRule.WebClient wc = j.createWebClient();
-        wc.login("alice", "alice");
-        HtmlForm form = wc.getPage(p, "configure").getFormByName("config");
-        form.getInputByName("pseudoRemoteTrigger").setChecked(true);
-        form.getInputByName("authToken").setValueAttribute("secret");
-        j.submit(form);
-        hudson.model.BuildAuthorizationToken token = p.getAuthToken();
-        assertNotNull(token);
-        assertEquals("secret", token.getToken());
-        wc = j.createWebClient();
         wc.assertFails(p.getUrl() + "build?token=secret", HttpURLConnection.HTTP_FORBIDDEN);
         j.waitUntilNoActivity();
         assertEquals(0, p.getBuilds().size());
         Page page = wc.goTo("buildByToken/build?job=" + p.getFullName() + "&token=secret&delay=0sec", null);
-        j.waitUntilNoActivity();
+        assertCreated(page);
         assertEquals(1, p.getBuilds().size());
-        assertEquals(HttpStatus.SC_CREATED, page.getWebResponse().getStatusCode());
-        assertTrue(page.getWebResponse().getResponseHeaderValue("Location").contains("/queue/item/"));
         page = wc.goTo("buildByToken/build?job=" + p.getFullName() + "&token=secret&delay=0sec", null);
-        j.waitUntilNoActivity();
+        assertCreated(page);
```

Comment:
Do not recall now. I suppose just to verify that each web request produces a build. Anyway that existed before this PR.